### ISSUE TITLE
Failure in creating the worker node - bootstrap provider not ready. 

### ISSUE THREAD

**Issue by harishspqr** [kubernetes-sigs/cluster-api-provider-aws#1373]
[capi_logs_create_worker_failure.txt](https://github.com/kubernetes-sigs/cluster-api-provider-aws/files/3880962/capi_logs_create_worker_failure.txt)
[capa_logs_create_worker_failure.txt](https://github.com/kubernetes-sigs/cluster-api-provider-aws/files/3880963/capa_logs_create_worker_failure.txt)
[capbk_logs_create_worker_failure.txt](https://github.com/kubernetes-sigs/cluster-api-provider-aws/files/3880964/capbk_logs_create_worker_failure.txt)

/kind bug

**What steps did you take and what happened:**
Create a cluster using cluster API

**What did you expect to happen:**
We expect the cluster to be created however It was noticed that the control plane was created however the worker failed to be created. 
From CAPA logs: 
```
I1121 17:09:01.444533 1 machine_controller.go:170] Reconciliation for Machine "worker-awscluster1-default-node-pool-5f4bfd88cc-n7h7b" in namespace "80fc77f71f1a9c4c0972443223a23a5a799c2e7b" asked to requeue: `Bootstrap provider for Machine "worker-awscluster1-default-node-pool-5f4bfd88cc-n7h7b" in namespace "80fc77f71f1a9c4c0972443223a23a5a799c2e7b" is not ready`, requeuing: requeue in 30s
Following up with upstream cluster API team for more insights.
```
From capbk logs: 
```
I1121 16:34:00.793980 1 kubeadmconfig_controller.go:110] KubeadmConfigReconciler "level"=0 "msg"="Waiting for Machine Controller to set OwnerRef on the KubeadmConfig" "kubeadmconfig"={"Namespace":"80fc77f71f1a9c4c0972443223a23a5a799c2e7b","Name":"worker-awscluster1-default-node-pool-lwnkr"}
I1121 16:34:00.794007 1 controller.go:242] controller-runtime/controller "level"=1 "msg"="Successfully Reconciled" "controller"="kubeadmconfig" "request"={"Namespace":"80fc77f71f1a9c4c0972443223a23a5a799c2e7b","Name":"worker-awscluster1-default-node-pool-lwnkr"}
I1121 16:34:00.848250 1 kubeadmconfig_controller.go:110] KubeadmConfigReconciler "level"=0 "msg"="Waiting for Machine Controller to set OwnerRef on the KubeadmConfig" "kubeadmconfig"={"Namespace":"80fc77f71f1a9c4c0972443223a23a5a799c2e7b","Name":"worker-awscluster1-default-node-pool-lwnkr"}
I1121 16:34:00.848276 1 controller.go:242] controller-runtime/controller "level"=1 "msg"="Successfully Reconciled" "controller"="kubeadmconfig" "request"={"Namespace":"80fc77f71f1a9c4c0972443223a23a5a799c2e7b","Name":"worker-awscluster1-default-node-pool-lwnkr"} 
I1122 18:47:32.026711       1 kubeadmconfig_controller.go:189] KubeadmConfigReconciler "level"=0 "msg"="Machine is not a control plane. If it should be a control plane, add `cluster.x-k8s.io/control-plane: true` as a label to the Machine" "kubeadmconfig"={"Namespace":"80fc77f71f1a9c4c0972443223a23a5a799c2e7b","Name":"worker-awscluster1-default-node-pool-lwnkr"} "machine-name"="worker-awscluster1-default-node-pool-5f4bfd88cc-n7h7b"
```
**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**
cluster-api-aws-controller:v0.4.4
cluster-api-kubeadm-controller:v0.1.3
kube-rbac-proxy:v0.4.0
cluster-api-controller:v0.2.5

- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

**Comment by ncdc** [comment]
The control plane never initialized successfully. This log line:

```
I1122 18:47:32.026711       1 kubeadmconfig_controller.go:189] KubeadmConfigReconciler "level"=0 "msg"="Machine is not a control plane. If it should be a control plane, add `cluster.x-k8s.io/control-plane: true` as a label to the Machine" "kubeadmconfig"={"Namespace":"80fc77f71f1a9c4c0972443223a23a5a799c2e7b","Name":"worker-awscluster1-default-node-pool-lwnkr"} "machine-name"="worker-awscluster1-default-node-pool-5f4bfd88cc-n7h7b"
```

indicates that the Kubeadm Boostrap provider is unable to process a worker machine at the moment, because it's still waiting on the control plane to be initialized.


In the capi pod logs, I see this repeatedly:

```
│E1122 18:49:29.125830       1 machine_controller_noderef.go:83] Failed to assign NodeRef to Machine "control-plane-awscluster1-0" in namespace "80fc77f71f1a9c4c0972443223a23a5a799c2e7b": Get https://awscluster1-apiserver-354819189.us-east-1.elb.amazonaws.com:6443/api/v1/nodes: EOF                                                                                                                                                                                                               │
```

Because of this, I decided to check on the AWSMachine for the first control plane machine in the capa logs. I saw this:

```
│I1121 16:27:37.655703       1 recorder.go:52] controller-runtime/manager/events "msg"="Warning"  "message"="Invalid update: Root volume size cannot be mutated from 0 to 50" "object"={"kind":"AWSMachine","namespace":"80fc77f71f1a9c4c0972443223a23a5a799c2e7b","name":"control-plane-awscluster1-0","uid":"2a0e750a-73d5-4796-b733-fbeac67ca946","apiVersion":"infrastructure.cluster.x-k8s.io/v1alpha2","resourceVersion":"250636175"} "reason"="InvalidUpdate"                                     ┤
```

And eventually, every time this AWSMachine is reconciled, it is skipped because it's in a terminally failed state:

```
│I1121 18:16:56.178291       1 awsmachine_controller.go:263] controllers/AWSMachine "msg"="Error state detected, skipping reconciliation" "awsCluster"="awscluster1" "awsMachine"="control-plane-awscluster1-0" "cluster"="awscluster1" "machine"="control-plane-awscluster1-0" "namespace"="80fc77f71f1a9c4c0972443223a23a5a799c2e7b"                                                                                                                                                                   │
```

I'm not sure what that failed state is - could you please `kubectl describe` and `kubectl get -o yaml` this AWSMachine?

**Comment by harishspqr** [comment]
```
 kubectl get machines
NAME                                                    PROVIDERID                    PHASE
control-plane-awscluster1-0                             aws:////i-<instanceID>   failed
worker-awscluster1-default                                                                        pending
```
Describing the failed machine:
```Name:         worker-awscluster1-default
Namespace:    80fc77f71f1a9c4c0972443223a23a5a799c2e7b
Labels:       cluster.x-k8s.io/cluster-name=awscluster1
              machine-template-hash=1906984477
Annotations:  <none>
API Version:  cluster.x-k8s.io/v1alpha2
Kind:         Machine
Metadata:
  Creation Timestamp:  2019-11-21T16:34:00Z
  Finalizers:
    machine.cluster.x-k8s.io
  Generate Name:  worker-awscluster1-default
  Generation:     1
  Owner References:
    API Version:           cluster.x-k8s.io/v1alpha2
    Block Owner Deletion:  true
    Controller:            true
    Kind:                  MachineSet
    Name:                  worker-awscluster1-default
    UID:                   4d7d16be-d364-4987-9a36-bbd225f8eca2
  Resource Version:        250691962
  Self Link:               /apis/cluster.x-k8s.io/v1alpha2/namespaces/80fc77f71f1a9c4c0972443223a23a5a799c2e7b/machines/worker-awscluster1-default
  UID:                     1cc2fa69-64d6-4f29-8cc3-c4bba2f87181
Spec:
  Bootstrap:
    Config Ref:
      API Version:  bootstrap.cluster.x-k8s.io/v1alpha2
      Kind:         KubeadmConfig
      Name:         worker-awscluster1-default
      Namespace:   
  Infrastructure Ref:
    API Version:  infrastructure.cluster.x-k8s.io/v1alpha2
    Kind:         AWSMachine
    Name:         worker-awscluster1-default
    Namespace:    80fc77f71f1a9c4c0972443223a23a5a799c2e7b
  Metadata:
  Version:  1.16.1
Status:
  Phase:  pending
Events:   <none>
```
Get Machine -o yaml
```apiVersion: cluster.x-k8s.io/v1alpha2
kind: Machine
metadata:
  creationTimestamp: "2019-11-21T16:34:00Z"
  finalizers:
  - machine.cluster.x-k8s.io
  generateName: worker-awscluster1-default
  generation: 1
  labels:
    cluster.x-k8s.io/cluster-name: awscluster1
    machine-template-hash: "1906984477"
  name: worker-awscluster1-default
  namespace:
  ownerReferences:
  - apiVersion: cluster.x-k8s.io/v1alpha2
    blockOwnerDeletion: true
    controller: true
    kind: MachineSet
    name: worker-awscluster1-default
    uid: 4d7d16be-d364-4987-9a36-bbd225f8eca2
  resourceVersion: "250691962"
  selfLink: /apis/cluster.x-k8s.io/v1alpha2/namespaces/80fc77f71f1a9c4c0972443223a23a5a799c2e7b/machines/worker-awscluster1-default
  uid: 1cc2fa69-64d6-4f29-8cc3-c4bba2f87181
spec:
  bootstrap:
    configRef:
      apiVersion: bootstrap.cluster.x-k8s.io/v1alpha2
      kind: KubeadmConfig
      name: worker-awscluster1-default
      namespace:
  infrastructureRef:
    apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
    kind: AWSMachine
    name: worker-awscluster1-default
    namespace:
  metadata: {}
  version: 1.16.1
status:
  phase: pending
```

**Comment by harishspqr** [comment]
AWS machine:
```
Name:         worker-awscluster1-default
Namespace:   
Labels:       <none>
Annotations:  <none>
API Version:  infrastructure.cluster.x-k8s.io/v1alpha2
Kind:         AWSMachine
Metadata:
  Creation Timestamp:  2019-11-21T16:34:00Z
  Finalizers:
    awsmachine.infrastructure.cluster.x-k8s.io
  Generate Name:  worker-awscluster1-default
  Generation:     1
  Owner References:
    API Version:     cluster.x-k8s.io/v1alpha2
    Kind:            Machine
    Name:            worker-awscluster1-default
    UID:             1cc2fa69-64d6-4f29-8cc3-c4bba2f87181
  Resource Version:  250691961
  Self Link:         /apis/infrastructure.cluster.x-k8s.io/v1alpha2/namespaces/80fc77f71f1a9c4c0972443223a23a5a799c2e7b/awsmachines/worker-awscluster1-default
  UID:               86589c92-dfb3-405c-9e63-7ca5f71946f7
Spec:
  Additional Tags:
  Ami:
    Id:                  ami-05999436bc48503b0
  Iam Instance Profile:  nodes
  Instance Type:         m5.large
  Root Device Size:      50
  Ssh Key Name:          default
Events:                  <none>
```
get machine -o yaml
```
+ kubectl get awsmachine -o yaml
apiVersion: v1
items:
- apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
  kind: AWSMachine
  metadata:
    annotations:
      cluster-api-provider-aws: "true"
      sigs.k8s.io/cluster-api-provider-aws-last-applied-tags: 
    creationTimestamp: "2019-11-21T16:27:18Z"
    finalizers:
    - awsmachine.infrastructure.cluster.x-k8s.io
    generation: 2
    labels:
      cluster.x-k8s.io/cluster-name: awscluster1
      cluster.x-k8s.io/control-plane: "true"
    name: control-plane-awscluster1-0
    namespace: 80fc77f71f1a9c4c0972443223a23a5a799c2e7b
    ownerReferences:
    - apiVersion: cluster.x-k8s.io/v1alpha2
      kind: Machine
      name: control-plane-awscluster1-0
      uid: 59d73852-8801-4f18-abe4-d979e8d15a99
    resourceVersion: "251547446"
    selfLink: /apis/infrastructure.cluster.x-k8s.io/v1alpha2/namespaces/80fc77f71f1a9c4c0972443223a23a5a799c2e7b/awsmachines/control-plane-awscluster1-0
    uid: 2a0e750a-73d5-4796-b733-fbeac67ca946
  spec:
    ami:
      id: ami-05999436bc48503b0
    iamInstanceProfile: control-plane
    instanceType: m5.large
    providerID: aws:////i-01544d418f4d0eac4
    rootDeviceSize: 50
    sshKeyName: tmc-default
  status:
    errorMessage: EC2 instance state "stopped" is unexpected
    errorReason: UpdateError
    instanceState: stopped
    ready: true
- apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
  kind: AWSMachine
  metadata:
    creationTimestamp: "2019-11-21T16:34:00Z"
    finalizers:
    - awsmachine.infrastructure.cluster.x-k8s.io
    generateName: worker-awscluster1-default
    generation: 1
    name: worker-awscluster1-default
    namespace:
    ownerReferences:
    - apiVersion: cluster.x-k8s.io/v1alpha2
      kind: Machine
      name: worker-awscluster1-default
      uid: 1cc2fa69-64d6-4f29-8cc3-c4bba2f87181
    resourceVersion: "250691961"
    selfLink: /apis/infrastructure.cluster.x-k8s.io/v1alpha2/namespaces/80fc77f71f1a9c4c0972443223a23a5a799c2e7b/awsmachines/worker-awscluster1-default
    uid: 86589c92-dfb3-405c-9e63-7ca5f71946f7
  spec:
    additionalTags:
    ami:
      id: ami-05999436bc48503b0
    iamInstanceProfile: nodes
    instanceType: m5.large
    rootDeviceSize: 50
    sshKeyName:default
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""
```

**Comment by ncdc** [comment]
```
    errorMessage: EC2 instance state "stopped" is unexpected
    errorReason: UpdateError
    instanceState: stopped
```

That's the problem

**Comment by ncdc** [comment]
Should be fixed by https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/1369. Not yet in a CAPA release.

**Comment by ncdc** [comment]
@vincepri sounds like we should do a new v0.4.x release

**Comment by harishspqr** [comment]
@ncdc Should we make it part of a release ?

**Comment by ncdc** [comment]
Fixed by v0.4.6
/close
